llvm/test/CodeGen/RISCV/poison-legalization.ll

```diff
@@ -0,0 +1,15 @@
+; RUN: llc < %s -mtriple=riscv32  | FileCheck %s
+
+define void @b(ptr %p1, ptr %p2)  {
+; CHECK:        .cfi_startproc
```

Comment:
```suggestion
define void @b(ptr %p1, ptr %p2) nounwind {
```

That should remove the .cfi_startproc.

Also I'd suggest one function for each type?